include/fastrtps/types/DynamicLoanableSequence.hpp

```diff
@@ -0,0 +1,213 @@
+#define _TYPES_DYNAMIC_LOANABLE_SEQUENCE_HPP_
+
+#include <cassert>
+#include <memory>
+
+#include <fastdds/dds/core/LoanableSequence.hpp>
+#include <fastdds/dds/log/Log.hpp>
+#include <fastrtps/types/DynamicData.h>
+#include <fastrtps/types/DynamicPubSubType.h>
+#include <fastrtps/types/DynamicTypePtr.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+/**
+ * @brief LoanableSequence specialization for DynamicData.
+ *
+ * This class provides a sequence container for handling loanable collections
+ * of DynamicData.
+ *
+ * @tparam _NonConstEnabler to enable data access through [] operator on non-const sequences.
+ */
+template<typename _NonConstEnabler>
+class LoanableSequence<fastrtps::types::DynamicData, _NonConstEnabler>
+    : public LoanableTypedCollection<fastrtps::types::DynamicData, _NonConstEnabler>
+{
+public:
+
+    /// Type for the size of the sequence.
+    using size_type = LoanableCollection::size_type;
+
+    /// Type for the elements in the sequence.
+    using element_type = LoanableCollection::element_type;
+
+    /**
+     * @brief Construct a LoanableSequence with a specified dynamic type.
+     *
+     * @param[in] dyn_type Pointer to the DynamicType.
+     */
+    LoanableSequence(
+            fastrtps::types::DynamicType_ptr dyn_type)
+        : dynamic_type_support_(new fastrtps::types::DynamicPubSubType(dyn_type))
+    {
+    }
+
+    /**
+     * @brief Construct a LoanableSequence with a specified maximum size.
+     *
+     * @param[in] max Maximum size of the sequence.
+     */
+    LoanableSequence(
+            size_type max)
+    {
+        if (max <= 0)
+        {
+            return;
+        }
+
+        resize(max);
+    }
+
+    /**
+     * @brief Destructor for LoanableSequence.
+     */
+    ~LoanableSequence()
+    {
+        if (elements_ && !has_ownership_)
+        {
+            EPROSIMA_LOG_WARNING(SUBSCRIBER, "Sequence destroyed with active loan");
+            return;
+        }
+
+        release();
+    }
+
+    /**
+     * @brief Copy constructor for LoanableSequence.
+     *
+     * @param[in] other The other LoanableSequence to copy from.
+     */
+    LoanableSequence(
+            const LoanableSequence& other)
+    {
+        *this = other;
+    }
+
+    /**
+     * @brief Copy assignment operator for LoanableSequence.
+     *
+     * @param[in] other The other LoanableSequence to assign from.
+     *
+     * @return A reference to this LoanableSequence.
+     */
+    LoanableSequence& operator =(
+            const LoanableSequence& other)
+    {
+        if (!has_ownership_)
+        {
+            release();
+        }
+
+        LoanableCollection::length(other.length());
+        const element_type* other_buf = other.buffer();
+        for (size_type n = 0; n < length_; ++n)
+        {
+            *static_cast<fastrtps::types::DynamicData*>(elements_[n]) =
+                    *static_cast<const fastrtps::types::DynamicData*>(other_buf[n]);
+        }
```

Comment:
This could copy `DynamicData` objects between collections constructed with different `DynamicType` values